projects/graph/src/graph.py

```diff
@@ -146,15 +146,25 @@ def bfs(self, start, goal):
         # Visited all paths; no connecting route
         return 'No connecting path exist'
 
+    def dfs(self, start, goal):
+        # Analogues to bfs but paths are now accessed using FILO order
+        visited = set()
+        paths = [[start]]
+        if start == goal:
+            return paths[0]
+        while paths:
+            path = paths.pop()
+            node = path[-1]
+            if node not in visited:
+                neighbors = self.vertices[node]
+                for neighbor in neighbors:
+                    new_path = list(path)
+                    new_path.append(neighbor)
+                    paths.append(new_path)
+                    if neighbor == goal:
+                        return new_path
+                    visited.add(node)
```

Comment:
Really nice and clean implementation of DFS.  The differences between DFS and BFS is not much, but it would be great if there were more comments added to describe the sequence of events, since they may not be so apparent to the code reviewer.  Great job though!